src/compiler/types.ts

```diff
@@ -6154,6 +6154,8 @@ export const enum TypeFlags {
     IncludesInstantiable = Substitution,
     /** @internal */
     NotPrimitiveUnion = Any | Unknown | Void | Never | Object | Intersection | IncludesInstantiable,
+    /** @internal */
+    NotStringLiteral = ~(StringLiteral | EnumLiteral | TemplateLiteral | StringMapping)

```

Comment:
`EnumLiteral` could be set for number enums, and is redundant, since `StringLiteral` is always also set on stringy-enums.

```suggestion
    NotStringLiteral = ~(StringLiteral | TemplateLiteral | StringMapping)

```